cl/alerts/management/commands/cl_send_alerts.py

```diff
@@ -216,13 +227,22 @@ def send_emails(self, rate):
                 # paired with a list of document dicts, of the form:
                 # [[alert1, [{hit1}, {hit2}, {hit3}]], [alert2, ...]]
                 if len(results) > 0:
-                    hits.append(
-                        [alert, qd.get("type", SEARCH_TYPES.OPINION), results]
-                    )
+                    search_type = qd.get("type", SEARCH_TYPES.OPINION)
+                    hits.append([alert, search_type, results])
                     alert.query_run = qd.urlencode()
                     alert.date_last_hit = now()
                     alert.save()
 
+                    # Send webhook event if the user has a SEARCH_ALERT
+                    # endpoint enabled.
+                    user_webhooks = user.webhooks.filter(
+                        event_type=WebhookEventType.SEARCH_ALERT, enabled=True
+                    )
+                    if user_webhooks.exists():
+                        self.send_search_alert_webhook(
+                            results, user_webhooks.first(), search_type, alert
+                        )
```

Comment:
From the docs:

> If the QuerySet has no ordering defined, then the queryset is automatically ordered by the primary key.

I guess your approach here, using `first()` would thus be mostly fine, because we enforce one enabled webhook per event type on the client side, but I could imagine us using the admin panel to create a second webhook for an event type for a particular client that asked us to.

So I think this might be better (and in the normal case would perform just as well):

```suggestion
                    for user_webhook in user_webhooks:
                        self.send_search_alert_webhook(
                            results, user_webhook, search_type, alert
                        )
```

(

Actually, in the case where a user only has one webhook, this would perform better, because you'd have one query:

    SELECT * FROM webhooks where user = x;

Instead of two:

    SELECT * FROM webhooks where user = x LIMIT 1;
    SELECT * FROM webhooks where user = x;

But this difference is very marginal! I just like to think carefully about our poor database.

)

Reply:
great! yes, this solves the problem if at any moment a user has more than one webhook endpoint.